### PR TITLE
remove SKlearnCompat

### PR DESCRIPTION
**Pull request by shinnar**
instead making Operators natively more scikit-learn compatible (with some hacks)

### CODE REVIEW

File: lale/operators.py [hunk 1]
```diff
@@ -701,6 +793,26 @@ def __getitem__(self, key: str) -> Any:  # enumeration.Enum:
             raise KeyError("No enumeration found for hyper-parameter: " + key)
 
 
+class WGP(object):
```
Comment (hirzel) [inline lale/operators.py:796]: Can you give this class a more descriptive name please? Also, should the class name start with an `_` to indicate it is not part of the public API?

File: lale/operators.py [hunk 2]
```diff
@@ -3446,6 +3721,25 @@ def __call__(self, steps: List[Any]):
 Pipeline = _PipelineFactory()
 
 
+def pipeline_graph_class(steps) -> Type[PlannedPipeline]:
```
Comment (hirzel) [inline lale/operators.py:3724]: Maybe add a leading underscore to this function name too.

File: lale/sklearn_compat.py [hunk 3]
```diff
@@ -49,9 +44,7 @@
 
 # This method (and the to_lale() method on the returned value)
 # are the only ones intended to be exported
-def make_sklearn_compat(
-    op: Union[Ops.Operator, "SKlearnCompatWrapper", Any]
-) -> "SKlearnCompatWrapper":
+def make_sklearn_compat(op):
```
Comment (hirzel) [inline lale/sklearn_compat.py:47]: Can you please add a comment explaining that this function is only there for backward compatibility and will be removed in a future release? In fact, we may even want to declare it deprecated.

**Review by hirzel — COMMENTED** [review]
Thanks a lot for these changes. They will make Lale simpler by removing a layer of indirection and corresponding more closely to what people who are familiar with sklearn would expect.

This PR does not seem to change the tests. Does that mean that the functionality was adequately covered by existing tests and those tests continue to work exactly as-is? May be worth to double-check if any new tests need to be added.

File: lale/operators.py (hunk 2 above)
Comment (shinnar) [reply lale/operators.py]: done

File: lale/sklearn_compat.py (hunk 3 above)
Comment (shinnar) [reply lale/sklearn_compat.py:22]: done

Comment (shinnar) [reply lale/sklearn_compat.py:22]: I also removed all calls to it from the lale code base

**Comment by shinnar** [comment]
> Thanks a lot for these changes. They will make Lale simpler by removing a layer of indirection and corresponding more closely to what people who are familiar with sklearn would expect.
> 
> This PR does not seem to change the tests. Does that mean that the functionality was adequately covered by existing tests and those tests continue to work exactly as-is? May be worth to double-check if any new tests need to be added.

The existing tests for sklearn_compat now run directly against the operator.  While more tests would be good, test coverage should be comparable with what was existing.

**Comment by shinnar** [comment]
I also added documentation to the operators module docstring.

**Pull request closed without merging**